bundler/lib/bundler/compact_index_client/parser.rb

```diff
@@ -69,15 +69,16 @@ def gem_parser
       # This is mostly the same as `split(" ", 3)` but it avoids allocating extra objects.
       # This method gets called at least once for every gem when parsing versions.
       def parse_version_checksum(line, checksums)
-        line.freeze # allows slicing into the string to not allocate a copy of the line
         name_end = line.index(" ")
+        return if name_end.nil? # Artifactory has a bug where blank lines exist in the versions file
         checksum_start = line.index(" ", name_end + 1) + 1
+        return if checksum_start.nil?
         checksum_end = line.size - checksum_start
-        # freeze name since it is used as a hash key
-        # pre-freezing means a frozen copy isn't created
-        name = line[0, name_end].freeze
+
+        line.freeze # allows slicing into the string to not allocate a copy of the line
+        name = line[0, name_end]
         checksum = line[checksum_start, checksum_end]
-        checksums[name] = checksum
+        checksums[name.freeze] = checksum # pre-freezing name means a frozen copy isn't created
```

Comment:
Just reshuffled a bit here:
* I moved `line.freeze` down so we don't freeze before returning.
* I also moved `name.freeze` into a spot that allowed me to explain with code, shortening the comment.
* I scratched my pedantic code nitpicking itch